setup.py

```diff
@@ -328,6 +328,15 @@ def get_requirements() -> List[str]:
     if _is_cuda():
         with open(get_path("requirements.txt")) as f:
             requirements = f.read().strip().split("\n")
+            cuda_major = torch.version.cuda.split(".")[0]
+            modified_requirements = []
+            for req in requirements:
+                if "vllm-nccl-cu12" in req:
+                    modified_requirements.append(
+                        req.replace("vllm-nccl-cu12",
+                                    f"vllm-nccl-cu{cuda_major}"))
+                else:
+                    modified_requirements.append(req)
```

Comment:
Shouldn't we add `requirements = modified_requirements`?

Reply:
Nice catch! Fixed in [fee6165](https://github.com/vllm-project/vllm/pull/3805/commits/fee61656e65cd8ce5dc5ce959f86782a71f7071b)